chemper/graphs/cluster_graph.py

```diff
@@ -262,6 +297,128 @@ def _add_first_smirks_atoms(self, mol, smirks_atoms):
                                          neighbor_storage,
                                          bond=bond_storage)
 
+        for smirks_index, atom_index in smirks_atoms.items():
```

Comment:
Seems like you should have comments here; not totally obvious what you're doing without thinking about it for a while/reading context.

Reply:
I was going to add more detail to the new functions, but this one could probably use more help too.